### PR TITLE
Do not call NPCReachedDestination() for soldiers without a MercProfile

### LINKED ISSUE #1657: Repeatable crash when opening tactical view in sector after game load
### What is wrong
If I try to open tactical view in B13 (Drassen Airport) in the attached savegame, the game crashes.

Interestingly, if I first open a different sector (I tried it with A2, Chitzena and whichever one in Gambria I have mercs in), then I can load B13 just fine.

| JA2S version | Game version        | Vanilla bug? | Used to work? |
| ------------ | ------------------- | ------------ | ------------- |
|  0.19.1 release | JA2 Gold Pack from Steam  |    ??    |    no idea |

**Log output**
```
2022-09-03T20:41:01 [INFO] externalized\ModPackContentManager.cc: Enabled mods                    'generous-rebels, from-russia-with-love, imp-quiz-honest-answers'
2022-09-03T20:41:02 [INFO] game\GameLoop.cc: Version Label: Stracciatella v0.19.1-git+6ad1b6f
2022-09-03T20:41:02 [INFO] game\GameLoop.cc: Version #:     Build 04.12.02
2022-09-03T20:41:24 [WARN] game\Tactical\Tactical_Save.cc: SynchronizeItemTempFile() Reported 0, should be 92
2022-09-03T20:41:26 [ERROR] sgp\SGP.cc: Game has been terminated due to an unrecoverable error: invalid m_profileID '200'
Creating an emergency savegame succeeded (Error.sav).
Please report this error with a description of the circumstances. Do not forget to attach the savegame. (class std::runtime_error)
2022-09-03T20:41:27 [ERROR] launcher\Launcher.cc: Command_execute "C:\\Program Files\\JA2 Stracciatella\\ja2.exe" []: exited with status code -1073740791
```

[2022-09-03t20-19-34z-crash.zip](https://github.com/ja2-stracciatella/ja2-stracciatella/files/9483362/2022-09-03t20-19-34z-crash.zip)

### PR DESCRIPTION
**Pull request by momoko-h**
Fixes #1657. Also fixes an OOB in the MercProfile constructor and cleans up struct SOLDIERTYPE a little.

### CODE REVIEW

**Comment by momoko-h** [comment]
Found another issue while looking at `NPCReachedDestination`'s other callers.

File: src/game/TacticalAI/NPC.cc [hunk 1]
```diff
@@ -1928,37 +1927,38 @@ void ConverseFull(UINT8 const ubNPC, UINT8 const ubMerc, Approach bApproach, UIN
...
+								//pNPC->uiStatusFlags &= ~SOLDIER_COWERING;
+								EVENT_InitNewSoldierAnim(pNPC, STANDING, 0 , FALSE);
+							}
 
-						pSoldier->ubQuoteRecord = ubRecordNum + 1; // add 1 so that the value is guaranteed nonzero
+							pNPC->ubQuoteRecord = ubRecordNum + 1; // add 1 so that the value is guaranteed nonzero
+						}
 
-						if (pQuotePtr->sActionData == NPC_ACTION_TELEPORT_NPC)
+						if (pNPC && pQuotePtr->sActionData == NPC_ACTION_TELEPORT_NPC)
```
Comment (lynxlynxlynx) [inline src/game/TacticalAI/NPC.cc:1956]: Looks like if `pNPC` is null, nothing will be done in the other branch as well, so this whole block can be moved inside the existing `if (pNPC...` instead.

**Comment by selaux** [comment]
@momoko-h Do you want to address lynx comment? Otherwise I think we can merge this.

**Comment by selaux** [comment]
I'll merge as the comment is less about functionality and more about style.

**Pull request merged**